gaphor/SysML/requirements/requirement.py

```diff
@@ -27,26 +28,30 @@ class RequirementItem(Classified, ElementPresentation[Requirement]):
     def __init__(self, diagram, id=None):
         super().__init__(diagram, id)
 
-        self.watch("show_stereotypes", self.update_shapes).watch(
-            "show_attributes", self.update_shapes
-        ).watch("show_operations", self.update_shapes).watch(
-            "subject[NamedElement].name"
-        ).watch("subject[NamedElement].namespace.name").watch(
-            "subject[Classifier].isAbstract", self.update_shapes
-        ).watch("subject[AbstractRequirement].externalId", self.update_shapes).watch(
-            "subject[AbstractRequirement].text", self.update_shapes
-        )
+        self.watch("show_text", self.update_shapes).watch(
+            "show_stereotypes", self.update_shapes
+        ).watch("show_attributes", self.update_shapes).watch(
+            "show_operations", self.update_shapes
+        ).watch("subject[NamedElement].name").watch(
+            "subject[NamedElement].namespace.name"
+        ).watch("subject[Classifier].isAbstract", self.update_shapes).watch(
+            "subject[AbstractRequirement].externalId", self.update_shapes
+        ).watch("subject[AbstractRequirement].text", self.update_shapes)
         attribute_watches(self, "Requirement")
         operation_watches(self, "Requirement")
         stereotype_watches(self)
 
+    show_text: attribute[int] = attribute("show_text", int, default=True)
+
     show_stereotypes: attribute[int] = attribute("show_stereotypes", int)
 
     show_attributes: attribute[int] = attribute("show_attributes", int, default=False)
 
     show_operations: attribute[int] = attribute("show_operations", int, default=False)
 
     def update_shapes(self, event=None):
+        # reset height so requirement block is reduced to minimum required height
+        self.height = RESET_HEIGHT
```

Comment:
I added this on a hunch, but I don't fully understand why this implementation works to always crop the box height to fit the text, would love to hear a brief explanation!

Reply:
Thanks for the clarification! Indeed, I thought about the intentions of users as well, but since resizing only happens when a user changes the text (or toggles the new button) I think this is acceptable for those users.